dummyserver/handlers.py

```diff
@@ -260,6 +262,17 @@ def status(self, request):
 
         return Response(status=status)
 
+    def retry_after(self, request):
+        if hasattr(self.application, 'last_req') and \
```

Comment:
Rather than use this `hasattr`, let's clean it up and ensure that the application always has a `last_req` field by setting it ourselves in `server.py`.